src/test/java/org/broadinstitute/hellbender/engine/ArgumentsBuilderWithJIMFS.java

```diff
@@ -0,0 +1,146 @@
+package org.broadinstitute.hellbender.engine;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+
+import htsjdk.samtools.reference.ReferenceSequenceFileFactory;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * ArgumentsBuilder specialization that includes management of a temporary JIMFS file system. This class
+ * can be used to test Path URIs on inputs that live on a java.nio file system, and are not java.io.File
+ * compatible.
+ *
+ * Methods accept File arguments, which are copied to a Jimfs file system managed by instacnes o this class,
+ * and Path URIs specifying the Jimfs copy are used as the actual argument values.
+ *
+ * TODO: Test methods need to declare IOExceptions since many of these methods throw IOException directly?
+ * TODO: This has to live in the test hierarchy in order to be able to import jimfs. Not sure why ArgumentsBuilder isn't also here
+ *
+ * Usage is intended to be via try-with-resources:
+ *
+ *     try (final ArgumentsBuilderWithJIMFS args = new ArgumentsBuilderWithJIMFS()) {
+ *         // add an input file with corresponding index
+ *         args.addFileWithIndexFromJIMFS(StandardArgumentDefinitions.INPUT_SHORT_NAME, cramFile, indexFile);
+ *
+ *         // add the reference, copying the reference, companion dictionary, and index to jimfs
+ *         args.addReferenceFileFromJIMFS(referenceFile);
+ *
+ *         final File outputFile = createTempFile("testReadCramWithIndexFromPath", ".bam");
+ *         args.add("-" + StandardArgumentDefinitions.OUTPUT_SHORT_NAME);
+ *         args.add(outputFile.getAbsolutePath());
+ *
+ *         runCommandLine(args);
+ *
+ *         // Assert.assert...
+ *     }
+ */
+public class ArgumentsBuilderWithJIMFS extends ArgumentsBuilder implements Closeable {
```

Comment:
Can you make the parent class implement `Closeable` as well, and make the jimfs initialization in this class lazy to prevent memory issues when using DataProviders?